unity-environment/Assets/ML-Agents/Scripts/Brain.cs

```diff
@@ -86,6 +89,15 @@ public class BrainParameters
  */
 public class Brain : MonoBehaviour
 {
+    // Current agent info
+    public Dictionary<int, List<float>> currentStates = new Dictionary<int, List<float>>();
+    public Dictionary<int, List<Camera>> currentCameras = new Dictionary<int, List<Camera>>();
+    public Dictionary<int, float> currentRewards = new Dictionary<int, float>();
+    public Dictionary<int, bool> currentDones = new Dictionary<int, bool>();
+    public Dictionary<int, float[]> currentActions = new Dictionary<int, float[]>();
+    public Dictionary<int, float[]> currentMemories = new Dictionary<int, float[]>();
+
+
```

Comment:
I'm going to make some comments regarding the coding convention for C# at Unity. You don't need to refactor all your code to the standard, but it's a good idea to progressively adhere to it. Alot of the coding convention is annoying and I find arbitrary, but some of it is useful because it enforces good patterns for APIs

Member fields (like currentStates, etc) should:

- always start with m_ /w camelcase (m_CurrentStates)

- never be public.  This is important because the API is not clear. For example, if I have an instance of a brain, can I call currentStates.Add(...)? Usually, you make your fields private or protected, use [SerializedField] if they need to persist, and make a matching property with relevant access rights.
Example:

```
protected Dictionary<int,int> m_CurrentStates = new ...
public IEnumerable<KeyValuePair<int,int>> states { get { return m_CurrentStates;} }

```